File: backend/wypisy/src/main/java/wypisy/example/wypisy/model/MachineProgram.java
package wypisy.example.wypisy.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotEmpty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Slf4j
public class MachineProgram {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotEmpty
    private String nameMachine;
    @NotEmpty
    private String nrProgram;
    private String description;
    private BigDecimal qtyForProgram;


    @ManyToMany(mappedBy = "machinePrograms")
    @JsonIgnore
    private List<ManufacturingProcess> processList=new ArrayList<>();








}

File: frontend/src/app/models/machineProgram.ts
export class MachineProgram {
    public id:number;
    public nameMachine: string;
    public nrProgram: string;
    public description: string;
    public qtyForProgram: number;


    constructor(
        id:number,
        nameMachine: string,
        nrProgram: string,
        description: string,
        qtyForProgram: number,

    ){
        this.id=id
        this.nameMachine=nameMachine
        this.nrProgram=nrProgram
        this.description=description
        this.qtyForProgram=qtyForProgram
    }

}